### PR TITLE
Feature/np list pagination

### PR DESCRIPTION
**Pull request by Vearso**

### CODE REVIEW

File: src/components/Products/ListPagination.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <div class="np-pagination__wrapper">
+    <Arrow @click="prevPage()"
+           class="np-pagination__wrapper__left-arrow"/>
+
+    <label for="pageNumber">
+      <input @change="e => setPageNumber(e.target.value)"
```
Comment (PietruszkaB) [inline src/components/Products/ListPagination.vue]: attribute order

File: src/components/Products/ListPagination.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    const nextPage = () => store.commit('products/nextPage');
+    const prevPage = () => store.commit('products/prevPage');
+    const setPageNumber = (page: number) => {
+      store.commit('products/setPage', page);
+      currentPage = computed(()=> store.getters['products/getPage']);
+    }
+
+    const filteredList: ComputedRef<Array<FoodListItem>> = computed(() => store.state.nav.filteredFoodList);
+    const setResults = (list: Array<FoodListItem>) => store.commit('products/setResults', list);
```
Comment (PietruszkaB) [inline src/components/Products/ListPagination.vue]: missing types

File: src/store/modules/products.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    setPage(state: ProductsState,page: number) {
+        state.pageNumber = 0;
+        if(page <= state.numberOfPages && page > 0) {
+            state.pageNumber = page;
+        } else state.pageNumber = 1;
+    },
+    setResults(state: ProductsState, list: Array<FoodListItem>) {
+        state.products = [];
+        for (let i = (state.pageNumber * 9 - 9); i < (state.pageNumber * 9); i++) {
```
Comment (PietruszkaB) [inline src/store/modules/products.ts]: i'd suggest creating a variable for number of products displayed on one page

**Review by PietruszkaB — CHANGES_REQUESTED** [review]

File: src/components/Products/ListPagination.vue [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  components: {
+    Arrow,
+  },
+  setup() {
+    const store = useStore();
+
+    let currentPage = computed(() => store.state.products.pageNumber);
+
+
+    const nextPage = () => store.commit('products/nextPage');
```
Comment (KamilWozniak) [inline src/components/Products/ListPagination.vue]: (): void

and same below

File: src/components/Products/ListPagination.vue [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    const prevPage = () => store.commit('products/prevPage');
+    const setPageNumber = (page: number) => {
+      store.commit('products/setPage', page);
+      currentPage = computed(()=> store.getters['products/getPage']);
```
Comment (KamilWozniak) [inline src/components/Products/ListPagination.vue]: why `currentPage` sometimes is a `store.state.products.pageNumber` and sometimes `store.getters['products/getPage']`?

if `products/setPage` modifies `store.state.products.pageNumber` (don't know if it does but it definitely should) then `currentPage` from 34 line will be automatically updated and there is no need to reassign that value as you did in line 41

File: src/components/Products/ListPagination.vue (hunk 4 above)
Comment (KamilWozniak) [reply src/components/Products/ListPagination.vue]: in multiple lines. check this out

File: src/components/Products/ListPagination.vue (hunk 1 above)
Comment (KamilWozniak) [reply src/components/Products/ListPagination.vue]: As we figured this out

File: src/store/modules/products.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            if(list[i] !== undefined) {
```
Comment (KamilWozniak) [inline src/store/modules/products.ts]: do you handle null as well? If not then just `if (list[i])` will be enough

File: src/store/modules/products.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                state.products.push(list[i]);
+            }
+        }
+    },
+    setNumberOfPages(state: ProductsState, list: Array<FoodListItem>) {
+        if (list.length % 9 !== 0) {
```
Comment (KamilWozniak) [inline src/store/modules/products.ts:49]: I see bare `9` all over the file. I assume that this is max number of items per page. What if suddenly client will want 12 items per page? Put all `9` in one variable and store it along with other global variables

File: src/store/modules/products.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            state.pageNumber += 1;
+        }
+    },
+    prevPage(state: ProductsState) {
+        if (state.pageNumber > 1) {
+            state.pageNumber -= 1
+        }
+    },
+    setPage(state: ProductsState,page: number) {
+        state.pageNumber = 0;
```
Comment (KamilWozniak) [inline src/store/modules/products.ts]: delete

**Review by KamilWozniak — CHANGES_REQUESTED** [review]

File: src/components/Products/ListPagination.vue [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    const nextPage = (): void => {
+      store.commit('products/nextPage');
+      inputValue.value = currentPage.value;
+    }
+    const prevPage = (): void => {
+      store.commit('products/prevPage');
+      inputValue.value = currentPage.value;
+    }
+
+    const setPage = (event: any): void => {
```
Comment (KamilWozniak) [inline src/components/Products/ListPagination.vue:44]: write custom interface

**Review by KamilWozniak — APPROVED** [review]

**Pull request merged**